packages/itwinui-react/src/core/Alert/Alert.tsx

```diff
@@ -6,26 +7,135 @@ import cx from 'classnames';
 import * as React from 'react';
 import '@itwin/itwinui-css/css/alert.css';
 import { useTheme, StatusIconMap, SvgCloseSmall } from '../utils/index.js';
-import type { CommonProps } from '../utils/props.js';
+import type {
+  CommonProps,
+  PolymorphicComponentProps,
+  PolymorphicForwardRefComponent,
+} from '../utils/props.js';
 
-export type AlertProps = {
+// ----------------------------------------------------------------------------
+// Alert.Icon component
+
+type AlertIconOwnProps = {
   /**
-   * Type of the alert.
-   * @default 'informational'
+   * Type of the alert icon.
    */
   type?: 'positive' | 'warning' | 'negative' | 'informational';
+};
+
+const AlertIcon = React.forwardRef((props, ref) => {
+  const { as: Element = 'svg', children, className, type, ...rest } = props;
+
+  if (type) {
```

Comment:
type should only be passed into the top-level `Alert` and it should be detected inside AlertIcon using context (instead of requiring passing it as a prop again)

Reply:
This has been changed. Users who pass `Alert.Icon` without any children will get the icon that corresponds with the `type` prop passed into `Alert`